plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubWebhook.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { InputError } from '@backstage/errors';
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { Octokit } from '@octokit/rest';
+import { parseRepoUrl } from '../publish/util';
+import { createTemplateAction } from '../../createTemplateAction';
+
+type ContentType = 'form' | 'json';
+
+export function createGithubWebhookAction(options: {
+  integrations: ScmIntegrationRegistry;
+}) {
+  const { integrations } = options;
+
+  const credentialsProviders = new Map(
+    integrations.github.list().map(integration => {
+      const provider = GithubCredentialsProvider.create(integration.config);
+      return [integration.config.host, provider];
+    }),
+  );
+
+  return createTemplateAction<{
+    repoUrl: string;
+    webhookUrl: string;
+    webhookSecret?: string;
+    events?: string[];
+    active?: boolean;
+    contentType?: ContentType;
+    insecureSsl?: boolean;
+  }>({
+    id: 'github:webhook',
+    description: 'Creates webhook for a repository on GitHub.',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl', 'webhookUrl'],
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            description: `Accepts the format 'github.com?repo=reponame&owner=owner' where 'reponame' is the new repository name and 'owner' is an organization or username`,
+            type: 'string',
+          },
+          webhookUrl: {
+            title: 'Webhook URL',
+            description: 'The URL to which the payloads will be delivered',
+            type: 'string',
+          },
+          webhookSecret: {
+            title: 'Webhook Secret',
+            description: 'Webhook secret value',
+            type: 'string',
+          },
+          events: {
+            title: 'Triggering Events',
+            description:
+              'Determines what events the hook is triggered for. Default: push',
+            type: 'array',
+            items: {
+              type: 'string',
```

Comment:
I'd probably suggest that we make a followup with an enum here to so that you can only choose the events you want, like the github UI rather than freetext. It's pretty error prone.

Reply:
I agree - it is going to be a bit more complicated - as in GitHub:
- push only (default)
- all (['*'])
- selected (get all possible events using GitHub API).

I am going to do this in a separate PR and I will look into how to make a default UI for it.

Reply:
I think you can do this using `oneOf` in `react-jsonschema-form` there's some guides here: https://react-jsonschema-form.readthedocs.io/en/latest/usage/oneof/

Reply:
You can have one of either, `push only` `*` or `selected` and then on the selected view you can add nested properties for each one

Reply:
Happy to help support if you get stuck, just reach out here or in DIscord and I can help.

Reply:
@benjdlambert  Thank you for your suggestions! 😃 I will contact you if I get stuck.

Reply:
@benjdlambert I have created a change with a validation: https://github.com/backstage/backstage/pull/7077. Although it required also changing https://github.com/octokit/webhooks.js.